packages/astro/src/vite-plugin-astro-server/route.ts

```diff
@@ -288,6 +288,12 @@ export async function handleRoute({
 				incomingResponse,
 			});
 	}
+
+	// We remove the internally-used header before we send the response to the user agent.
+	if (response.headers.has(REROUTE_DIRECTIVE_HEADER)) {
+		response.headers.delete(REROUTE_DIRECTIVE_HEADER);
+	}
+
```

Comment:
I wonder, should we also do this during the build? If so, at this point we should have this code inside the pipeline, because it seems we have this logic in SSR too:

https://github.com/withastro/astro/blob/87a3d51f2ca8661babbb76956e54bf389eb86d8f/packages/astro/src/core/app/index.ts#L323-L325

Of course, we can land this possible improvement in another PR

Reply:
You are exactly right! It makes sense for this to be managed by `Pipeline`/`RenderContext`.

Error page rerouting is currently the jurisdiction of `vite-plugin-astro-server` and `App`. When it is turns into a middleware, the header could be removed in a single place. It would be similar to how `X-Astro-Route-Type` is handled after the i18n middleware has read it.

Regarding doing this for static builds, I'm not sure if it would have an effect since headers are ignored altogether.